src/rgw/rgw_rest_s3.cc

```diff
@@ -1909,47 +1909,19 @@ int RGWPostObj_ObjStore_S3::get_policy()
   return 0;
 }
 
-int RGWPostObj_ObjStore_S3::complete_get_params()
-{
-  bool done;
-  do {
-    struct post_form_part part;
-    int r = read_form_part_header(&part, &done);
-    if (r < 0)
-      return r;
-
-    bufferlist part_data;
-    bool boundary;
-    uint64_t chunk_size = s->cct->_conf->rgw_max_chunk_size;
-    r = read_data(part.data, chunk_size, &boundary, &done);
-    if (!boundary) {
-      return -EINVAL;
-    }
-
-    parts[part.name] = part;
-  } while (!done);
-
-  return 0;
-}
-
 int RGWPostObj_ObjStore_S3::get_data(bufferlist& bl)
 {
   bool boundary;
   bool done;
 
-  uint64_t chunk_size = s->cct->_conf->rgw_max_chunk_size;
+  const uint64_t chunk_size = s->cct->_conf->rgw_max_chunk_size;
   int r = read_data(bl, chunk_size, &boundary, &done);
-  if (r < 0)
+  if (r < 0) {
     return r;
+  }
 
   if (boundary) {
     data_pending = false;
-
-    if (!done) {  /* reached end of data, let's drain the rest of the params */
```

Comment:
I agree we should not pass the arguments but I think we should read all the data